### PR TITLE
Add work with orders

### PR DESCRIPTION
**Pull request by VoitseshukV**

### CODE REVIEW

File: src/main/java/com/bookstore/core/model/Order.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "orders")
+@SQLDelete(sql = "UPDATE orders SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/com/bookstore/core/model/Order.java]:
```suggestion
@SQLDelete(sql = "UPDATE orders SET is_deleted = TRUE WHERE id=?")
@Where(clause = "is_deleted = FALSE")
```

File: src/main/java/com/bookstore/core/service/impl/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Override
+    @Transactional
+    public OrderDto updateOrderById(Long orderId, UpdateOrderDto requestDto) {
+        Order order = orderRepository.findFirstById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with ID: " + orderId));
+        order.setStatus(Order.OrderStatus.valueOf(requestDto.status()));
+        orderRepository.save(order);
+        return null;
```
Comment (okuzan) [inline src/main/java/com/bookstore/core/service/impl/OrderServiceImpl.java]: returning null is a bad practice, return updated Order

File: src/main/java/com/bookstore/core/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+        order.setTotal(total);
+        orderRepository.save(order);
+        shoppingCart.setCartItems(new HashSet<>());
+    }
+
+    @Override
+    public OrderDto getOrderById(String email, Long orderId) {
+        User user = userRepository.findByEmail(email).orElseThrow(
+                () -> new EntityNotFoundException("Can't get user with email: " + email));
```
Comment (okuzan) [inline src/main/java/com/bookstore/core/service/impl/OrderServiceImpl.java]: use service method for find/orElseThrow logic everywhere (create if needed)

File: src/main/java/com/bookstore/core/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setPrice(cartItem.getBook().getPrice());
+            order.getOrderItems().add(orderItem);
+            total = total.add(orderItem.getPrice().multiply(
+                    BigDecimal.valueOf(orderItem.getQuantity())));
+            cartItemRepository.delete(cartItem);
+        }
```
Comment (okuzan) [inline src/main/java/com/bookstore/core/service/impl/OrderServiceImpl.java]: method is too bulky, delegate some logic to private methods

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookstore/core/repository/order/OrderRepository.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.Optional;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface OrderRepository extends JpaRepository<Order, Long> {
+    List<Order> findAllByUser(User user, Pageable pageable);
+
+    Optional<Order> findFirstByUserAndId(User user, Long id);
+
+    Optional<Order> findFirstById(Long id);
```
Comment (Sanderfuck) [inline src/main/java/com/bookstore/core/repository/order/OrderRepository.java]: Is it possible find second byId, by id exist must only one

**Review by Sanderfuck — APPROVED** [review]

**Pull request merged**